modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/CheckboxTag.java

```diff
@@ -0,0 +1,192 @@
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.frontend.taglib.clay.internal.servlet.taglib.BaseContainerTag;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.taglib.util.TagResourceBundleUtil;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Kresimir Coko
+ */
+public class CheckboxTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		return super.doStartTag();
+	}
+
+	public boolean getChecked() {
+		return _checked;
+	}
+
+	public boolean getDisabled() {
+		return _disabled;
+	}
+
+	public boolean getIndeterminate() {
+		return _indeterminate;
+	}
+
+	public boolean getInline() {
+		return _inline;
+	}
+
+	public String getLabel() {
+		return _label;
+	}
+
+	public String getName() {
+		return _name;
+	}
+
+	public void setChecked(boolean checked) {
+		_checked = checked;
+	}
+
+	public void setDisabled(boolean disabled) {
+		_disabled = disabled;
+	}
+
+	public void setIndeterminate(boolean indeterminate) {
+		_indeterminate = indeterminate;
+	}
+
+	public void setInline(boolean inline) {
+		_inline = inline;
+	}
+
+	public void setLabel(String label) {
+		_label = label;
+	}
+
+	public void setName(String name) {
+		_name = name;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_checked = false;
+		_disabled = false;
+		_indeterminate = false;
+		_inline = false;
+		_label = null;
+		_name = null;
+	}
+
+	@Override
+	protected String getHydratedModuleName() {
+		return "frontend-taglib-clay/Checkbox";
+	}
+
+	@Override
+	protected Map<String, Object> prepareProps(Map<String, Object> props) {
+		props.put("checked", _checked);
+		props.put("disabled", _disabled);
+		props.put("indeterminate", _indeterminate);
+		props.put("inline", _inline);
+
+		if (Validator.isNotNull(_label)) {
+			props.put(
+				"label",
+				LanguageUtil.get(
+					TagResourceBundleUtil.getResourceBundle(pageContext),
+					_label));
+		}
+
+		props.put("name", _name);
+
+		return super.prepareProps(props);
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("custom-control");
+		cssClasses.add("custom-checkbox");
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		jspWriter.write("<label><input");
+
+		if (_checked) {
+			jspWriter.write(" checked");
+		}
+
+		jspWriter.write(" class=\"custom-control-input ");
+
+		if (Validator.isNotNull(getCssClass())) {
+			jspWriter.write(getCssClass());
+		}
+
+		jspWriter.write("\"");
+
+		if (_disabled) {
+			jspWriter.write(" disabled");
+		}
+
+		if (Validator.isNotNull(getId())) {
+			jspWriter.write(" id=\"");
```

Comment:
Similar problem here, we're likely going to have duplicated id's. 

This might not be obvious because you probably kill the output later with the React component, so try with JS disabled to see the real server output.

Reply:
Good idea, the turning off JS thing, will do!

Reply:
So from my investigation, turning off JS reveals that both the wrapper and the input element have the same `id="labeledCheckbox"`:
```html
<div class="custom-checkbox custom-control" id="labeledCheckbox">
	 <label>
	 	 <input class="custom-control-input " id="labeledCheckbox" name="checkboxWithLabel" type="checkbox">

	 	 <span class="custom-control-label">
	 		 <span class="custom-control-label-text">
	 	 	  	 Checkbox with Label
	 	 	  </span>
	 	 </span>
	 </label>
</div>
```

Not sure how to fix this, as I only have one instance of `getId()` in my tag, which is on the `input` tag. The `input` is the one that's supposed to have the `id`.

Reply:
This is probably something that we will have to deal in more tags.

One solution would be overwrite the entire `processStartTag` from `BaseContainerTag.java` in your tag, which seems not to be the best option.

I'm thinking about adding a method `isRenderIdInMainElement()` (or sth similar) to `BaseContainerTag.java`:

```
protected void isRenderIdInMainElement() {
   return true;
}
```

that you can overwrite in you `CheckboxTag.java`
```
protected void isRenderIdInMainElement() {
   return false;
}
```

and add this change to `BaseContainerTag.java`
```
		if (Validator.isNotNull(getId())) {
			jspWriter.write(" id=\"");
			jspWriter.write(getId());
			jspWriter.write("\"");
		}
```

to 

```
		if (isRenderIdInMainElement && Validator.isNotNull(getId())) {
			jspWriter.write(" id=\"");
			jspWriter.write(getId());
			jspWriter.write("\"");
		}
```

What do you guys think @jbalsas @kresimir-coko ?

Reply:
New method sounds better IMO, just the name feels a bit clunky, maybe something like `hasIdInBaseContainer`?

Reply:
See # first. After that, if we still want to go with this, which we might want to, we probably can do:

```java
protected void renderIdAttribute() {
	if (Validator.isNotNull(_id)) {
		jspWriter.write(" id=\"");
		jspWriter.write(_id);
		jspWriter.write("\"");
	}
}
```

That way we simply need to override that method in our `CheckboxTag` class to avoid rendering it in the container without any additional flags. It's more or less the same approach but we avoid needing stacking flags in the base component.

Reply:
Sounds good to me